runtime/call_metrics.go

```diff
@@ -234,3 +243,54 @@ func NewOutboundTChannelMetrics(scope tally.Scope) *OutboundTChannelMetrics {
 	metrics.SystemErrors = scope.Counter(outboundCallsSystemErrors)
 	return &metrics
 }
+
+// NewOutboundCustomMetrics returns outbound metrics for custom clients
+func NewOutboundCustomMetrics(scope tally.Scope) *OutboundCustomMetrics {
+	metrics := OutboundCustomMetrics{}
+	metrics.Sent = scope.Counter(outboundCallsSent)
+	metrics.Latency = scope.Timer(outboundCallsLatency)
+	metrics.Success = scope.Counter(outboundCallsSuccess)
+	metrics.Errors = scope.Counter(outboundCallsErrors)
+	return &metrics
+}
+
+// SetupCustomClientLoggersMetrics creates:
+// - client logger/scope
+// - client method loggers/metrics
+func SetupCustomClientLoggersMetrics(
+	clientID string,
+	methods []string,
+	deps *DefaultDependencies,
+) (
+	logger *zap.Logger,
+	scope tally.Scope,
+	loggers map[string]*zap.Logger,
+	metrics map[string]*OutboundCustomMetrics,
+) {
+	logger = deps.Logger.With(zap.String("clientID", clientID))
```

Comment:
just curious why `clientID -> clientID` and `client -> clientID` :)

Reply:
Originally, I had `client` as a field and tag key, but was asked to follow this convention of adding `ID` or `Name` to the key for logger fields.